packages/slate-editor/src/modules/nodes-hierarchy/queries/isDescendantOf.ts

```diff
@@ -1,10 +1,13 @@
-import { Node } from 'slate';
+import { Editor } from 'slate';
 
 import type { HierarchyNodeQuery } from '../types';
 
 export function isDescendantOf(parentMatch: HierarchyNodeQuery): HierarchyNodeQuery {
-    return (_, path, editor) => {
-        for (const [parentNode, parentPath] of Node.levels(editor, path)) {
+    return (node, path, editor) => {
+        for (const [parentNode, parentPath] of Editor.levels(editor, {
+            at: path,
+            match: (n) => n !== node,
+        })) {
```

Comment:
Let's untangle this a little bit:

```suggestion
        const levels = Editor.levels(editor, {
            at: path,
            match: (n) => n !== node,
        });
        for (const [parentNode, parentPath] of levels) {
```